Content.Server/_NF/ShuttleRecords/ShuttleRecordsSystem.cs

```diff
@@ -73,6 +78,103 @@ record = component.ShuttleRecords[uid];
+
+        // sort through the records and use originalname to categorise ships
+        foreach (var record in records.ShuttleRecords.Values)
+        {
+            if (record.OriginalName is null)
+                continue;
+
+            if (!shipTypes.ContainsKey(record.OriginalName))
+                shipTypes.Add(record.OriginalName, new RecordSummary());
+
+            if (shipTypes.TryGetValue(record.OriginalName, out var value))
+            {
+                value.Count += 1;
+                totalShips += 1;
+
+                if (EntityManager.TryGetEntity(record.EntityUid, out _)) // check if the ship still exists
+                {
+                    value.AbandonedCount += 1;
+                    totalAbandoned += 1;
+                }
+
+                if (record.TimeOfPurchase is not null && record.TimeOfSale is not null)
+                {
+                    var lifetime = record.TimeOfSale.Value.Subtract(record.TimeOfPurchase.Value);
+                    value.Lifetimes.Add(lifetime);
+                    totalLifetimes.Add(lifetime);
+                }
+            }
+        }
+
+        var sortedSummaries = shipTypes.OrderByDescending(record => record.Value.Count).ThenBy(record => record.Key);
+
+        // export raw data as a file for discord
+
+        var rawData = JsonSerializer.SerializeToUtf8Bytes(shipTypes, new JsonSerializerOptions { WriteIndented = true, IncludeFields = true });
+
+        /* eventual discord message should be of the format
+        ```
+         Num │ Abnd │ Avg time │ Type
+        ─────┼──────┼──────────┼───────────
+        1234 │ 1234 │    00:00 │ NAME
+        1234 │ 1234 │    00:00 │ NAME
+        1234 │ 1234 │    00:00 │ NAME
+        ─────┼──────┼──────────┼───────────
+        1234 │ 1234 │    00:00 │
+        ```
+        */
+
+        builder.AppendLine("```");
+        builder.AppendLine(" Num │ Abnd │ Avg time │ Type");
+        builder.AppendLine("─────┼──────┼──────────┼───────────");
+        foreach (var record in sortedSummaries)
+        {
+            // fallback, in case every ship of this type was abandoned this round and there are no lifetimes to report
+            var averageLifetime = "N/A";
+            if (record.Value.Lifetimes.Count != 0)
+            {
+                averageLifetime = TimeSpan.FromSeconds(record.Value.Lifetimes.Average(timeSpan => timeSpan.TotalSeconds)).ToString(@"hh\:mm");
+            }
+
+            // pad data for formatting
+            builder.AppendLine($"{record.Value.Count,4} | {record.Value.AbandonedCount,4} |{averageLifetime,9} | {record.Key}");
+        }
+
+        builder.AppendLine("─────┼──────┼──────────┼───────────");
+
+        // fallback, in case somehow every single ship was abandoned this round and there are no lifetimes to report
+        var totalAvgLifetime = "N/A";
+        totalAvgLifetime = TimeSpan.FromSeconds(totalLifetimes.Average(timeSpan => timeSpan.TotalSeconds)).ToString(@"hh\:mm");
```

Comment:
```suggestion
        if (totalLifetimes.Count > 0)
            totalAvgLifetime = TimeSpan.FromSeconds(totalLifetimes.Average(timeSpan => timeSpan.TotalSeconds)).ToString(@"hh\:mm");
```
Avoid getting the average of an empty list, that'll throw an InvalidOperationException going by https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.average?view=net-9.0